packages/block-editor/src/hooks/gap.js

```diff
@@ -109,6 +113,13 @@ export function GapEdit( props ) {
 		setAttributes( {
 			style: cleanEmptyObject( newStyle ),
 		} );
+
+		// In Safari, changing the `gap` CSS value on its own will not trigger the layout
+		// to be recalculated / re-rendered. To force the updated gap to re-render, here
+		// we replace the block's node with itself.
+		if ( ref.current ) {
```

Comment:
Is there a way to limit it to only Safari cause it sounds a bit aggressive :)

Reply:
Very good point! I'll add in some user agent sniffing tomorrow 🙂

Reply:
@youknowriad I've added in some user agent sniffing here, consistent with [this MDN article](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) for detecting Safari (string contains "Safari" but doesn't contain "Chrome" or "Chromium").

It appears to work okay in my testing. I thought of abstracting the check out to somewhere else (perhaps in a package somewhere), but it doesn't look like we're using an explicit UA check for Safari anywhere else, so I thought I'd leave it inline here for now, and until we have a second use for it somewhere. But, do let me know if you can think of a better way to handle the check. Thanks!